### DISCUSSION TITLE
PingImage doesnt work on EPS file on windows

### DISCUSSION THREAD

**Discussion by serban-seeq** [ImageMagick/ImageMagick#4493] [Help]
Im running IM 7.0.8 with GS(ghostscript) 9.55.0 on windows 64 bit

I'm calling PingImage. It eventually gets to an InvokePostscriptDelegate call where the command passed in is `"gs" -q -dQUIET -dSAFER -dBATCH -dNOPAUSE -dNOPROMPT -dMaxBitmap=500000000 -dAlignToPixels=0 -dGridFitTT=2 "-sDEVICE=pngalpha" -dTextAlphaBits=%u -dGraphicsAlphaBits=%u "-r%s" %s "-sOutputFile=%s" "-f%s" "-f%s"` 

call stack
>	IM_MOD_RL_ps_.dll!ReadPSImage(const _ImageInfo * image_info, _ExceptionInfo * exception) Line 842	C
 	CORE_RL_MagickCore_.dll!ReadImage(const _ImageInfo * image_info, _ExceptionInfo * exception) Line 543	C
 	CORE_RL_MagickCore_.dll!ReadStream(const _ImageInfo * image_info, unsigned __int64(*)(const _Image *, const void *, const unsigned __int64) stream, _ExceptionInfo * exception) Line 1043	C
 	CORE_RL_MagickCore_.dll!PingImage(const _ImageInfo * image_info, _ExceptionInfo * exception) Line 268	C

I then get an exception with the following reason:
`FailedToExecuteCommand "gs" -q -dQUIET -dSAFER -dBATCH -dNOPAUSE -dNOPROMPT -dMaxBitmap=500000000 -dAlignToPixels=0 -dGridFitTT=2 "-sDEVICE=pngalpha" -dTextAlphaBits=4 -dGraphicsAlphaBits=4 "-r72x72" -g686x860 -dEPSCrop  "-sOutputFile=C:/Users/serbanm/AppData/Local/Temp/magick-22264YWzu4nyewcfW%d" "-fC:/Users/serbanm/AppData/Local/Temp/magick-22264pZTR9idMoOGp" "-fC:/Users/serbanm/AppData/Local/Temp/magick-22264f2tvEU8DHAv_"' (The system cannot find the file specified.
) @ error/delegate.c/ExternalDelegateCommand/459`

The command seems to be coming from `delegates.xml`

Now I noticed in the GS bin folder there is no gs.exe file. As soon as I rename gswin64.exe => gs.exe the issue is resolved.

Is this the expected behavior? Convert.exe is able to resolve the exe filename without me renaming so I was expecting a call to PingImage would as well. What is the proper configuration change that I need in order to have this work on win64?

**Comment by snibgo** [comment]
What is your delegates.xml entry for decoding EPS? Mine (v7.1.0-4 on Windows 8.1) is:
```
  <delegate decode="eps" encode="pdf" mode="bi" command="&quot;@PSDelegate@&quot; -q -dQUIET -dSAFER -dBATCH -dNOPAUSE -dNOPROMPT -dMaxBitmap=500000000 -sDEVICE=pdfwrite &quot;-sOutputFile=%o&quot; -- &quot;%i&quot;"/>
```
Note "@PSDelegate@". IM tries to get the path to Ghostscript from the Windows registry. I suspect the problem on your computer is that there is a mismatch between the registry and the Ghostscript software. Try resinstalling IM or GS or both.

**Reply by serban-seeq** [reply]
@snibgo when I run `identify.exe -list delegate` I get 

```
Path: built-in

Delegate                Command
-------------------------------------------------------------------------------
...
        eps<=>ps        "gs" -q -dQUIET -dSAFER -dBATCH -dNOPAUSE -dNOPROMPT -dMaxBitmap=500000000 -dAlignToPixels=0 -dGridFitTT=2 "-sDEVICE=ps2write" "-sOutputFile=%o" "-f%i"
        eps<=>pdf       "gs" -q -dQUIET -dSAFER -dBATCH -dNOPAUSE -dNOPROMPT -dMaxBitmap=500000000 -dAlignToPixels=0 -dGridFitTT=2 "-sDEVICE=pdfwrite" "-sOutputFile=%o" "-f%i"
...
```


Note that they are built-in and there is no delegates.xml in my output directory when I run convert.exe, but its using the delegates.xml from the source to generate this output folder on inside delegates.xml I have
`<delegate decode="eps" encode="pdf" mode="bi" command="&quot;gs&quot; -sstdout=%%stderr -dQUIET -dSAFER -dBATCH -dNOPAUSE -dNOPROMPT -dMaxBitmap=500000000 &quot;-sDEVICE=pdfwrite&quot; &quot;-sOutputFile=%o&quot; &quot;-f%i&quot;"/>
  `
This is the same with what is currently in the main branch. I guess I should also mentioned I dont actually have imagemagick installed, just the source code downloaded and used in my application.

**Comment by snibgo** [comment]
> I guess I should also mentioned I dont actually have imagemagick installed, just the source code downloaded and used in my application.

Okay, I think that explains it. You haven't installed IM, so IM hasn't put an entry in the registry. As you realise, IM tries to run "gs.exe", so you need to rename or copy or create a symbolic link to "gswin64.exe".

On my Windows 8.1 laptop, "gs.exe" is on my system path but "gswin64" isn't.
